### PR TITLE
XArrayInterface improvements: dimension units and labels

### PR DESCRIPTION
**Pull request by drs251**
This addresses the first two points in #2319. As @philippjfr suggested, it's relatively straightforward to read units and dimension labels from xarray and attach them to the appropriate key and value dimensions (as far as I can tell, the attributes `units` and `long_name` are a standard for netcdf files). I could not reproduce the the problem with the swapped kdims for xarray.DataArray vs. xarrayDataset, but I added a test to guard against it.

I would like to try to add some stuff to this PR with your input (see #2319):
- ~~Support for dimensions without coordinates~~
- ~~Perhaps a better way to deal with non-coordinate dimensions~~
- [x] When everything else is done: some extra documentation.

### CODE REVIEW

**Comment by philippjfr** [comment]
Looks good so far and the other suggestions sound good too although I suspect it might get tricky.

**Comment by drs251** [comment]
I'm tempted to agree; the first two points should probably be implemented together, and the second one could be a real head scratcher. I'll try to put some thought into it, but if it's too ambitious, I'll just add the documentation I had in mind and leave the issue open for another time. Anyway, I would like to get this in the next release, if possible. Any ideas when that will be?

**Comment by drs251** [comment]
OK, then I'll probably concentrate on the docs for now.

**Comment by drs251** [comment]
@philippjfr OK, I added a simple example of how visualize an xarray Dataset. I also found that something similar already exists for GeoViews, so I added some links. I don't think I'll have time right now to work on the other ideas I have for XArrayInterface, so I would consider this PR to be finished, but I'll leave issue #2319 open and hopefully I'll manage to come back to it sometime.

**Comment by philippjfr** [comment]
At a quick glance this looks great! Would you mind clearing the notebook and committing that. We generally don't store notebooks with output and it will make it easier to review.

File: examples/user_guide/08-Gridded_Datasets.ipynb [hunk 1]
```diff
@@ -292,6 +292,69 @@
...
+    "%%opts Image [colorbar=True]\n",
+    "%%output size=200\n",
+    "hv_ds.to(hv.Image, kdims=[\"lon\", \"lat\"])[:, 220:320, :]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note, that this approach immediately converts all available data to images, which will take up a lot of RAM for large datasets. For these situations, consider using a [DynamicMap](./06-Live_Data.ipynb) in conjunction with [xarray's dask support](http://xarray.pydata.org/en/stable/dask.html) instead.\n",
```
Comment (philippjfr) [inline examples/user_guide/08-Gridded_Datasets.ipynb]: Maybe mention the ``dynamic`` argument to ``.to`` here, it will generate a DynamicMap for you.

**Comment by philippjfr** [comment]
One minor comment, otherwise looks good to merge.

**Comment by philippjfr** [comment]
You mean changing a label after it has already been set? It was a deliberate decision not to allow that although @jlstevens was more insistent on that than I was. You can override the existing label like this though:

```python
xr_ds = xr.tutorial.load_dataset("air_temperature")
hv.Dataset(xr_ds, vdims=('air', 'air_temp')).redim(air=('air', 'air_temperature')).vdims[0].label
```

**Comment by drs251** [comment]
Sorry, but I still don't get it: I'm absolutely unable to replace the rather long-winded "4xDaily Air temperature at sigma level 995" axis label in the plot within Holoviews. On the other hand, it's not essential for this PR though.

**Comment by philippjfr** [comment]
Right, that's a separate issue in any case. I'll play around once merged and if I can reproduce I'll raise an issue.

**Pull request merged**